image-rs/src/image.rs

```diff
@@ -274,15 +274,22 @@ impl ImageClient {
         let reference = Reference::try_from(image_url)
             .map_err(|source| PullImageError::IllegalImageReference { source })?;
 
-        let tasks = match &self.registry_handler {
-            Some(handler) => handler
+        let tasks = if let Some(config) = &self.config.registry_config {
+            let registry_handler = RegistryHandler::new(config.clone()).unwrap();
+            registry_handler
                 .process(reference)
-                .map_err(|source| PullImageError::IllegalRegistryConfigurationFormat { source })?,
-            None => vec![ImagePullTask {
-                image_reference: reference,
-                use_http: false,
-                task_type: TaskType::Origininal,
-            }],
+                .map_err(|source| PullImageError::IllegalRegistryConfigurationFormat { source })?
+        } else {
+            match &self.registry_handler {
+                Some(handler) => handler.process(reference).map_err(|source| {
+                    PullImageError::IllegalRegistryConfigurationFormat { source }
+                })?,
+                None => vec![ImagePullTask {
+                    image_reference: reference,
+                    use_http: false,
+                    task_type: TaskType::Origininal,
+                }],
+            }
```

Comment:
The registry handler is initialized when the ImageClient is being created. You might not need to change this file, but add some logic near https://github.com/confidential-containers/guest-components/blob/main/image-rs/src/builder.rs#L149-L158 . 

- If `registry_config` is given in the config, use this config and skip `registry_configuration_uri` field
- If `registry_config` is not given, use `registry_configuration_uri`